sa/sa.go

```diff
@@ -1277,6 +1277,71 @@ func (ssa *SQLStorageAuthority) FQDNSetExists(ctx context.Context, names []strin
 	return count > 0, err
 }
 
+// PreviousCertificateExists returns true iff there was at least one certificate
+// issued with the provided domain name, and the most recent such certificate
+// was issued by the provided registration ID. Note: This means that if two
+// different accounts were issuing certificates for a domain, only one gets the
+// right to revalidate using TLS-SNI-01. We think this is an acceptable tradeoff
+// of complexity versus coverage, though we may reconsider in the future.
+func (ssa *SQLStorageAuthority) PreviousCertificateExists(
+	ctx context.Context,
+	req *sapb.PreviousCertificateExistsRequest,
+) (*sapb.Exists, error) {
+	t := true
+	exists := &sapb.Exists{Exists: &t}
+
+	f := false
+	notExists := &sapb.Exists{Exists: &f}
+
+	// Find the most recently issued certificate containing this domain name.
+	var serials []string
+	_, err := ssa.dbMap.Select(
```

Comment:
This should be a `SelectOne`, then we won't have to mess around with slices below.

Reply:
I think this was a choice to avoid `SelectOne` due to the Gorp err squash issue?

Reply:
It was. However, on thinking about it more, this workaround is not really necessary. If we get a timeout, which leads to a `sql: no rows in result set` error, and then turn that into a 500, that's actually what we want (though the logging may be a little wonky). This code would turn that into "huh, no previous certificate, so I guess you have to do the http-01 challenge." Which might cause people more breakage than a 500. I'll back out the SelectOne -> Select changes.